src/ring_theory/finiteness.lean

```diff
@@ -260,10 +274,30 @@ end
 /-- If `f : A →ₐ[R] B` is surjective with finitely generated kernel and `A` is finitely presented,
 then so is `B`. -/
 lemma of_surjective {f : A →ₐ[R] B} (hf : function.surjective f) (hker : f.to_ring_hom.ker.fg)
-  (hfp : finitely_presented R A) : finitely_presented R B :=
+  (hfp : finite_presentation R A) : finite_presentation R B :=
 equiv (quotient hker hfp) (ideal.quotient_ker_alg_equiv_of_surjective hf)
 
-end finitely_presented
+/-- An algebra is finitely presented if and only if it is a quotient of a polynomial ring whose
+variables are indexed by a fintype by a finitely generated ideal. -/
+lemma iff_quotient_mv_polynomial' : finite_presentation R A ↔ ∃ (ι : Type) [fintype ι]
+  (f : (_root_.mv_polynomial ι R) →ₐ[R] A), (surjective f) ∧ f.to_ring_hom.ker.fg :=
+begin
+  split,
+  { rintro ⟨n, ⟨f, hf⟩⟩,
+    use [fin n, by apply_instance, f, hf] },
```

Comment:
The start of the `Type u_2` proof could be:
```
  { rintro ⟨n, ⟨f, hf⟩⟩,
    let e : ulift (fin n) ≃ fin n := ⟨ulift.down, ulift.up, ulift.up_down, ulift.down_up⟩,
    refine ⟨ulift (fin n), by apply_instance, f.comp (mv_polynomial.rename e),
      hf.1.comp _, _⟩,
    { simp only [], sorry, },
    { change (f.to_ring_hom.comp _).ker.fg,
      rw [ring_hom.ker_eq_comap_bot, ←ideal.comap_comap, ←ring_hom.ker_eq_comap_bot],
      sorry}, },
```

Annoyingly there doesn't seem to be a statement that `mv_polynomial.rename e` is bijective when `e` is, but maybe I just didn't look in the right place.

Reply:
Thank you! I've modified it using your suggestion. It can probably be golfed a little, I am not familiar at all with `ulift`.